blocks/library/gallery/block.js

```diff
@@ -74,7 +92,8 @@ class GalleryBlock extends Component {
 	}
 
 	onSelectImages( imgs ) {
-		this.props.setAttributes( { images: imgs } );
+		const images = imgs.map( img => img.caption ? { ...img, caption: [ img.caption ] } : img );
+		this.props.setAttributes( { images } );
```

Comment:
Minor: I would probably use something more descriptive like:

```js
onSelectImages( images ) {
	this.props.setAttributes( {
		images: images.map( ( attributes ) => ( {
			...attributes,
			caption: attributes.caption || [],
		} ),
	} );
}
```

Additionally, why `[ img.caption ]`? Shouldn't `img.caption` be an array?

Reply:
I followed your suggestion and used most of your code, thank you :) 
Regarding this part:
> Shouldn't img.caption be an array?

This function executes when the images are selected from the media library, and we receive a string from it. That's why we wrap in an array so it is ready to be used in the editable.